packages/ui/src/collapsible/collapsible.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FC, useState, useEffect, useCallback } from "react";
+import { Disclosure, Transition } from "@headlessui/react";
+
+export type TCollapsibleProps = {
+  title: string | React.ReactNode;
+  children: React.ReactNode;
+  buttonClassName?: string;
+  isOpen?: boolean;
+  onToggle?: () => void;
+  defaultOpen?: boolean;
+};
+
+export const Collapsible: FC<TCollapsibleProps> = (props) => {
+  const { title, children, buttonClassName, isOpen, onToggle, defaultOpen } = props;
+  // state
+  const [localIsOpen, setLocalIsOpen] = useState<boolean>(isOpen || defaultOpen ? true : false);
+
+  useEffect(() => {
+    if (isOpen !== undefined) {
+      setLocalIsOpen(isOpen);
+    }
+  }, [isOpen]);
+
+  // handlers
+  const handleOnClick = useCallback(() => {
+    if (isOpen !== undefined) {
+      if (onToggle) onToggle();
+    } else {
+      setLocalIsOpen((prev) => !prev);
+    }
```

Comment:
```suggestion
    if (onToggle) onToggle();
    if (isOpen === undefined) {
      setLocalIsOpen((prev) => !prev);
    }
```
I think we can still trigger `onToggle` even when `isOpen` is `undefined`, so that we can have some side effect on Toggle without passing down `isOpen`

@SatishGandham what do you think?